### ISSUE TITLE
Incorrect metadata query in getMetadataTriggers method of MetadataTriggerService

### ISSUE THREAD

**Issue by matej-me** [trailheadapps/apex-recipes#238]
### Summary

In the custom metadata query in getMetadataTriggers method of MetadataTriggerService class there's currently an issue with the query incorrectly retrieving custom object API names. The issue lies in line 28.

```
SELECT Class__c
            FROM Metadata_Driven_Trigger__mdt
            WHERE
->             Object__c = :this.objType
                AND Enabled__c = TRUE ...
```

Should be changed to

```
SELECT Class__c
            FROM Metadata_Driven_Trigger__mdt
            WHERE
->            Object__r.QualifiedApiName = :this.objType
                AND Enabled__c = TRUE ...
```

Addressing Object__c field, which is of type Metadata Relationship(Entity Definition), will only correctly retrieve Trigger Object's API name when using standard objects. When using custom objects, it will return the ID of CustomEntityDefinition.

this.objType will always be the API name of the object, which can be retrieved by selecting the QualifiedApiName instead.

### Salesforce Org Type

Developer Edition Org

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by codefriar** [comment]
@matej-me when I execute the query, as written, it works just fine for me. Can you help me understand a bit more about what issues you're having with the query as written?

**Comment by matej-me** [comment]
@codefriar Sure. 

In the Custom Metadata Query about, this.objType is obtained by the call

Trigger.new[0].getSObjectType().getDescribe().getName();

Let's assume this is a custom object called Visit. The code above will return "Visit__c".

Now, let's run the following metadata query, assuming we have two metadata entries, one for a standard Contact object, another for the custom Visit object

SELECT Class__c, Object__c
FROM Metadata_Driven_Trigger__mdt

The result will be the following.

<html>
<body>


Class__c | Object__c
-- | --
ContactTriggerHandler | Contact
VisitTriggerHandler | 01I7a000000WWCg

</body>
</html>

The query as is currently will be matching "Visit__c" against "01I7a000000WWCg", because the Custom Metadata Entity Relationship field does not return a string with the API Name of the object when referenced directly.

Let's now add Object__r.QualifiedApiName in the query

<html>
<body>


Class__c | Object__c | Object__r.QualifiedApiName
-- | -- | --
ContactTriggerHandler | Contact | Contact
VisitTriggerHandler | 01I7a000000WWCg | Visit__c

</body>
</html>


Basically, the query will return nothing, because it will be matching the wrong data. It won't find an entry for our Trigger Handler class.

**Comment by codefriar** [comment]
@matej-me 

Ok, see this is where I'm confused. Here's a simplified query I'm testing with.
```
SELECT Class__c
FROM Metadata_Driven_Trigger__mdt
WHERE
    Object__c = 'Account'
    AND Enabled__c = TRUE
ORDER BY Execution_Order__c
```

I run this in an org where Apex Recipes is installed and it correctly returns two rows:

Here's the Log:
```
sfdx force:data:soql:query --query SELECT Class__c FROM Metadata_Driven_Trigger__mdt WHERE     Object__c = 'Account'     AND Enabled__c = TRUE ORDER BY Execution_Order__c
Querying Data... done
CLASS__C
──────────────────────────────
MDTAccountTriggerHandler
MDTSecondAccountTriggerHandler
Total number of records retrieved: 2.

```

I've not tried this with other objects, and I'm not opposed to your proposed change, I just want to make sure I understand the problem first and from my testing ... I don't see a problem.

**Comment by matej-me** [comment]
@codefriar Try adding a Metadata_Driven_Trigger__mdt record for a custom object. This will recreate the issue.

**Comment by codefriar** [comment]
So it does. Thanks for the pr and for helping me understand! I’m away from my computer atm, but I’ll get this submitted later today

**Comment by matej-me** [comment]
Not a problem. Hope to have been of service.
